tools/ci/appveyor/appveyor_build.sh

```diff
@@ -55,25 +55,26 @@ function set_configuration {
         mingw)
             build='--build=i686-pc-cygwin'
             host='--host=i686-w64-mingw32'
+	    dep='--disable-dependency-generation'
         ;;
         msvc)
             build='--build=i686-pc-cygwin'
             host='--host=i686-pc-windows'
+	    dep='--disable-dependency-generation'
         ;;
         msvc64)
             build='--build=x86_64-unknown-cygwin'
             host='--host=x86_64-pc-windows'
+	    dep='--enable-dependency-generation'
```

Comment:
Maybe it could be nice to have a comment to clarify that the point is to have at least one system where dependency-generation is checked?

(I'm a bit lost because I assumed that it was not doable on msvc, but doable on mingw (so I expected the first rather than third configuration to enable it). I suppose that the CI machines in fact also always have mingw around for that part? But then do we have a msvc-without-mingw test somewhere?)

Reply:
Good idea, given the regression - done!

Reply:
I have a meta point here: you are trying very hard to disable "all the useless things" for each test to reduce CI times (which I think is very nice), and yet you want each interesting aspect of the system to be tested at least once. This is a consistency invariant that is checked nowhere, automatically or even manually (you "just" try to be careful when you change things), and discipline failure are very hard to notice before they get revealed by a bug. I wonder if we could improve on this.

For example, one possible approach would be to ask for a justification comment for each `--disable-foo` switch that is passed somewhere, where the justification would explain why this is not needed and also which other parts of the CI system tests it. Another would be to have a central document that details all things that are purposefully checked, with a pointer to the check.